examples/preprocessing/eog_regression.py

```diff
@@ -65,9 +63,7 @@
 evoked_after = epochs_after.average()
 
 fig, ax = plt.subplots(nrows=3, ncols=2, figsize=(10, 7),
-                       sharex=True, sharey='row')
+                       sharex=True, sharey='row', layout='constrained')
```

Comment:
argh, this will require MPL >=3.4 I think. Our current "old/compat" CIs test with MPL 3.1 (and hence is failing on this line, and the similar `layout='tight'` line in `mne/viz/topomap.py`)

@larsoner AFAICT we don't specify a hard minimum version of MPL for users; what is our decision procedure for which version of MPL is the "soft" limit (used in our compat tests)?  For reference:

- 3.4.0: 2021 Mar 26 (17 months ago)
- 3.3.0: 2020 Jul 16 (25 months ago)
- 3.2.0: 2020 Mar 03 (29 months ago)
- 3.1.0: 2019 May 17 (39 months ago)

Reply:
>  what is our decision procedure for which version of MPL is the "soft" limit (used in our compat tests

We try to maintain 2 year compat based on the `X.Y.0` release date

> Our current "old/compat" CIs test with MPL 3.1 (and hence is failing on this line, and the similar layout='tight' line in mne/viz/topomap.py)

This inline comment is in an example, where `constrained_layout` / `layout='constrained'` should be fine to use. (We don't officially guarantee that our *examples* always work on the oldest dependencies, though we should try to be at least somewhat reasonable there, too). In the tests / main code it could be a problem, though.

Reply:
> We try to maintain 2 year compat based on the `X.Y.0` release date

ok, so we could in theory bump `compat/old` to use 3.3.0.  That wouldn't fix this problem though.

> This inline comment is in an example

quite right, the failure is due to `layout='tight'` in `topomap.py`

Reply:
I can just drop the `layout='tight'` in `topomap.py`, try to be less lazy and add the offsets myself with `adjust_subplots`